### ISSUE TITLE
FRED: Blank arguments in SEXPs will cause them to be removed during save, potentially causing broken events

### ISSUE THREAD

**Issue by AxemP** [scp-fs2open/fs2open.github.com#2314]
So make an Event

Make the condition "is-destroyed-delay"

Go to the Delay argument and make it blank. Just delete the number and don't put anything in.

If you try and ask FRED to detect errors, it will give none. So now save the mission, and FRED will eventually bring up the error (it might take a save or two)

```
Error in mission event "Event name": Argument count is illegal
In sexpression: ( when 
   ( is-destroyed-delay "GTF Ulysses 0" ) 
   ( do-nothing ) 
)

(Error appears to be: is-destroyed-delay)
```

Not sure what the best preventative course of action is. Detect missing mandatory arguments and give them a default value? Make FRED bring this up before the save will eliminate the empty arguments?

**Comment by Goober5000** [comment]
The argument isn't actually missing; it's just 0-length.  Of course, when the mission is re-parsed, FRED doesn't realize it's 0-length and thinks it's missing.

I think I have a solution.

**Comment by Goober5000** [comment]
Fixed in #2323.
